### ISSUE TITLE
Document how to restart a Kafka stream based processing pipeline

### ISSUE THREAD

**Issue by drewhk** [akka/reactive-kafka#229]
I.e. how to wait on previous instance to properly stop and rematerialize the stream, etc.

**Comment by nafshartous** [comment]
I'd like to help and have a few questions.  How does one build and preview the docs ?

**Comment by patriknw** [comment]
@nafshartous Sorry for embarrassing long delay of this reply. Use `sbt paradox`, and then you can open `docs/target/paradox/site/index.html` in browser.

**Comment by nafshartous** [comment]
Does something like this capture the intent ?

```
//restart kafka consumer
object RestartKafkaConsumeExample extends ConsumerExample {
  def main(args: Array[String]): Unit = {
    // #restartConsumer
    def createAndRun(): Consumer.Control =
      Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
        .toMat(Sink.ignore)(Keep.left).run()

    val control = createAndRun()

    control.shutdown()

    Await.result(control.isShutdown, 5.seconds)

    // Start the stream again
    createAndRun()
    // #restartConsumer
  }
}
```

**Comment by nafshartous** [comment]
Will submit a new PR based on the discussion in

  https://github.com/akka/reactive-kafka/pull/275

**Comment by jacobg** [comment]
Thanks for documenting an approach to restarting the stream. If this approach is attempting to deal with failure in the Kafka consumer source, then I'm wondering why one of approaches documented for handling failures in Akka streams up-stream sources by just restarting the failed stage is not used?
http://doc.akka.io/docs/akka/2.5.4/scala/stream/stream-error.html

Is not sufficient to just restart the Kafka source stage using RestartSource or a supervision strategy that restarts?

Are there cases where an Akka Kafka source's underlying implementation can fail without propagating the failure into the stream?

**Comment by kciesielski** [comment]
@jacobg Looks like `RestartSource` is a pretty new thing, added in Akka 2.5. I think we may need both usages in the example, for Akka 2.4 and 2.5.

**Comment by kciesielski** [comment]
@nafshartous restarting is already documented in https://github.com/akka/reactive-kafka/blob/master/docs/src/main/paradox/errorhandling.md. I'm not sure why it's not visible in the main docs. @patriknw does that require some kind of redeployment to the docs?

**Comment by jacobg** [comment]
Thanks @kciesielski. Does akka kafka stream implement supervision strategy for kafka or other internal issues not related to specific message elements? i.e., will all exceptions go to supervision strategy which can then be directed to restart the stream?

**Comment by ennru** [comment]
The state of the docs has improved since: https://doc.akka.io/docs/akka-stream-kafka/current/errorhandling.html#restarting-the-stream-with-a-backoff-stage
